<?xml version="1.0"?>
<robot name="myfirst">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>


  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.19 0.19 0.09"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <link name="link_1">
    <visual>
      <geometry>
        <cylinder length="0.08" radius="0.095"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.045"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="base_to_link_1" type="continuous">
    <parent link="base_link"/>
    <child link="link_1"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0 0 0.045"/>
  </joint>

  <link name="link_2">
    <visual>
      <geometry>
        <box size="0.08 0.065 0.24"/>
      </geometry>
      <origin rpy="0 0.78535 0" xyz="0.06 0 0.0978"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="base_to_link_2" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="30" lower="-1" upper="1" velocity="1.0"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin rpy="0 0 0" xyz="0 0 0.098"/>
  </joint>

  <link name="link_3">
    <visual>
      <geometry>
        <box size="0.24 0.045 0.045"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.12 0 0.0225"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="base_to_link_3" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="30" lower="-1" upper="1" velocity="1.0"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin rpy="0 0 0" xyz="0.15 0 0.18"/>
  </joint>

</robot>

